src/Microsoft.AspNetCore.Razor/TagHelpers/HtmlAttributeValueStyle.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.TagHelpers
+{
+    public enum HtmlAttributeValueStyle
+    {
+        DoubleQuotes,
+        SingleQuotes,
```

Comment:
`SingleQuotes` appears only 15 times in this entire PR. In addition to needing more coverage in a couple of test classes, this suggests the generation tests should be expanded to cover more single-quoted cases. E.g. I don't think preallocated single-quoted attributes appear more than once.